sklearn/utils/estimator_checks.py

```diff
@@ -327,18 +329,55 @@ def _get_check_estimator_ids(obj):
 
 
 def _construct_instance(Estimator):
```

Comment:
Maybe we could add a keyword argument `return_multiple_instances=False` to make it explicit for the caller to return multiple instances?

- by default, with `eturn_multiple_instances=False` it would only return `Estimator(Ridge(), {"alpha": [0.1, 1]})` for the `SearchCV` and direct instances for other estimators and meta-estimators,

- with `return_multiple_instances=True`, it would return the pair `(Estimator(Ridge(), {"alpha": [0.1, 1]}), Estimator(LogisticRegression(), {"C": [0.1, 1]})` (and for other estimators it would just return a single element tuple for other estimators and meta estimators.